azure-pipelines.yaml

```diff
@@ -0,0 +1,121 @@
+        python.version: '3.6'
+      linux_py37:
+        imageName: 'ubuntu-16.04'
+        python.version: '3.7'
+  pool:
+    vmImage: $(imageName)
+
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '$(python.version)'
+      architecture: 'x64'
+
+  - script: brew install node@12
+    displayName: 'Install NodeJS'
+  - script: brew install yarn
+    displayName: 'Install yarn'
+  - script: python -m pip install --upgrade ipywidgets traitlets pytest
+    # pip installs release candidates by default:
+    #   https://github.com/pypa/pip/issues/4969
+    displayName: 'Install dependencies'
+
+  - script: python -m pip install --upgrade black
+    displayName: 'Install black (linter)'
+    condition: eq(variables['python.version'], '3.7')
+  - script: black --check .
+    displayName: 'Test Formating'
+    condition: eq(variables['python.version'], '3.7')
+
+  - script: python -m pip install --upgrade .
+    displayName: 'Test Install'
+    condition: ne(variables['Agent.OS'], 'Windows_NT')
+
+    # Build the python distribution from source
+  - script: python setup.py bdist_wheel
+    displayName: 'Build wheel'
+    condition: ne(variables['Agent.OS'], 'Windows_NT')
+
+  - bash: |
+      set -xeo pipefail
+      # Display log files if the build failed
+      echo "Dumping log files for failed build"
+      echo "----------------------------------"
+      for f in $(find $BUILD_REPOSITORY_LOCALPATH/build -name *.log);
+        do echo "------"
+            echo $f
+            echo "======"
+            cat $f
+        done;
+    condition: failed() # only run this job if the build step failed
+    displayName: "Print log files (failed build only)"
+
+# download the artifact and deploy it only if the build job succeeded
+- job: Deploy
+  pool:
+    vmImage: 'ubuntu-16.04'
+
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '3.7'
+      architecture: 'x64'
+
+  - script: 'pip install twine wheel setuptools setuptools_scm'
+    displayName: 'Install twine/wheel'
+    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags'))
+
+    # Build the python distribution from source
+  - script: python setup.py bdist_wheel
+    displayName: 'Build wheel'
+    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags'))
+
+  - task: TwineAuthenticate@1
+    displayName: 'Twine Authenticate'
+    inputs:
+      pythonUploadServiceConnection: 'pypi-upload'
+
+  - script: python -m twine upload -r "pypi-upload" --config-file $(PYPIRC_PATH) dist/*.whl
+    displayName: 'Upload to Pypi'
+    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags'))
+
+- job: Deploy_js
+  pool:
+    vmImage: 'ubuntu-16.04'
+
+  steps:
+  - task: NodeTool@0
+    inputs:
+      versionSpec: '12.x'
+  - script: |
+      yarn
+    displayName: 'Install dependencies'
+  - script: |
+      yarn build
+    displayName: 'Build'
+  - script: |
```

Comment:
The scripts here need the `condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags'))` protection

Reply:
Added the condition for both build and release

Reply:
Perfect thanks!